web/pdf_page_view.js

```diff
@@ -236,17 +238,16 @@ class PDFPageView {
 
     switch (state) {
       case RenderingStates.INITIAL:
+        this.#canvasWrapper?.classList.add("loadingIcon");
+        break;
```

Comment:
Why is this needed, shouldn't we still hide the loadingIcon until rendering is actually started?

Reply:
The `loadingIcon` is hidden by default:
https://github.com/mozilla/pdf.js/pull/15992/files#diff-e676e52ece9bec7287184fc8ab806d393ca42da7bb3105ac93b4b6cd0d23a3efR161

then adding/removing `loading` will trigger a visibility change.

Reply:
When rendering is cancelled we'll set RenderingStates.INITIAL and it seems that we definitely want to *ensure* that the loadingIcon is hidden in that case as well, hence I don't understand why these new lines are introduced here!?

Reply:
When the rendering is finished, the icon is removed but when the page is zoomed, reset is called and the state is set to INITIAL, loadingIcon is added but invisible and when RUNNING is set then `loading` is added and the icon is made visible.

Reply:
With this patch the `loadingIcon` now isn't added until the `draw` method is called, which can happen much later (or not at all) after the `reset` method was invoked.

Hence I don't understand why we simply don't want to *still* do the same thing in the INITIAL/PAUSED cases here!